src/werkzeug/datastructures.py

```diff
@@ -2952,6 +2952,11 @@ def save(self, dst, buffer_size=16384):
         from shutil import copyfileobj
 
         close_dst = False
+        if not PY2:
```

Comment:
This should be handled as an optional dependency until support for 3.5 is dropped:

```python
try:
    import pathlib
except ImportError:
    pathlib = None
```
```python
if pathlib is not None and isinstance(dst, pathlib.PurePath):
    dst = str(dst)
```